pkg/api/types.go

```diff
@@ -1290,6 +1290,32 @@ func (k *KubernetesConfig) IsRBACEnabled() bool {
 	return false
 }
 
+// UserAssignedIDEnabled checks if the user assigned ID is enabled or not.
+func (k *KubernetesConfig) UserAssignedIDEnabled() bool {
```

Comment:
Can you incorporate these new functions in existing unit tests if they aren't already getting covered?

Reply:
Sure @CecileRobertMichon . I will put out another PR to start doing that.

Reply:
what do you mean by "rationalize our test paths"?